### ISSUE TITLE
Fetch remote tags to local

### ISSUE THREAD

**Issue by manuwell** [src-d/go-git#371]
Hey guys, 

I'm trying to fetch remote tags but it seems that tags are ignored when fetching or pulling from remote. Trying to perform the following operations:

```go
         r.Fetch(&git.FetchOptions{
                 RemoteName: "origin",
         })

         refs, err := r.References()
         CheckIfError(err)

         err = refs.ForEach(func(ref *plumbing.Reference) error {
                 fmt.Println(ref)
                 return nil
         })
```

outputs
```
3dbebd24ff81327fd7171ccfa4d4ed81a6070069 refs/remotes/origin/master
3dbebd24ff81327fd7171ccfa4d4ed81a6070069 refs/heads/master
ref: refs/remotes/origin/master refs/remotes/origin/HEAD
ref: refs/heads/master HEAD
```

But no tags were fetched. What I'm trying to perform is a `git pull --tags` git operation. Is it already supported on `go-git`? 

Thanks and congrats for this project.

**Comment by smola** [comment]
I think we're doing the equivalent of `git fetch --no-tags` and you can fetch tags explicitely by adding the refspec `+refs/tags/*:refs/tags/*` to your fetch. We could look into being able to replicate the default git behavior though.

**Comment by manuwell** [comment]
I tried passing the suggested `refspec` but seems that object already exists. 

Debugging here I found this [piece of code](https://github.com/src-d/go-git/blob/727bf94da8e3cebd3ff467d30425b12d671fbca7/remote.go#L381) returning `true` to a fetched `tag` because, I think, it is looking for the commit hash instead the tag name to check the presence. 

The `storer` used on my local tests is `filesystem`. I couldn't go further in this problem.
